### PR TITLE
Bring back CMake install and uninstall as an unmaintained feature

### PR DESCRIPTION
**Pull request by Groovounet**

### CODE REVIEW

File: cmake/CMakePackageConfigHelpers.cmake [hunk 1]
```diff
@@ -0,0 +1,227 @@
+# - CONFIGURE_PACKAGE_CONFIG_FILE(), WRITE_BASIC_PACKAGE_VERSION_FILE()
```
Comment (fbridault) [inline cmake/CMakePackageConfigHelpers.cmake]: This file is part of [CMake modules](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html), why do you need a copy here ?

File: cmake/GNUInstallDirs.cmake [hunk 2]
```diff
@@ -0,0 +1,188 @@
+# - Define GNU standard installation directories
```
Comment (fbridault) [inline cmake/GNUInstallDirs.cmake]: This file is also part of [CMake modules](https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html), why do you need a copy here ?

File: glm/CMakeLists.txt [hunk 3]
```diff
@@ -42,8 +42,7 @@ source_group("SIMD Files" FILES ${SIMD_SOURCE})
 source_group("SIMD Files" FILES ${SIMD_INLINE})
 source_group("SIMD Files" FILES ${SIMD_HEADER})
 
-add_library(glm INTERFACE)
-target_include_directories(glm INTERFACE ../)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/..)
```
Comment (fbridault) [inline glm/CMakeLists.txt]: I would really recommend to use dependencies, and not use explicit include_directories. You can find the explanation in this [really great tutorial](https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/#model-dependencies-with-target_link_libraries).

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -231,7 +231,6 @@ function(glmCreateTestGTC NAME)
 	add_test(
 		NAME ${SAMPLE_NAME}
 		COMMAND $<TARGET_FILE:${SAMPLE_NAME}> )
-	target_link_libraries(${SAMPLE_NAME} PRIVATE glm::glm)
```
Comment (fbridault) [inline test/CMakeLists.txt]: I think this was the right way to link with glm (even if we only have headers). See https://pabloariasal.github.io/2018/02/19/its-time-to-do-cmake-right/#model-dependencies-with-target_link_libraries

File: cmake/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,66 @@
+##################################
+# CMake install / uninstall script
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
+set(GLM_INSTALL_CONFIGDIR "${CMAKE_INSTALL_LIBDIR}/cmake/glm")
+install(DIRECTORY glm DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```
Comment (neok-m4700) [inline cmake/CMakeLists.txt]: If I may, `make install` fails to install without the following changes:
```diff
--- cmake/CMakeLists.txt  2020-01-05 18:47:34.167576983 +0000
+++ cmake/CMakeLists.txt  2020-01-05 18:53:38.162324733 +0000
@@ -6,7 +6,7 @@
 include(CMakePackageConfigHelpers)
 
 set(GLM_INSTALL_CONFIGDIR "${CMAKE_INSTALL_LIBDIR}/cmake/glm")
-install(DIRECTORY glm DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/glm DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # CMake automatically adds an architecture compatibility check to make sure
 # 32 and 64 bit code is not accidentally mixed. For a header-only library this
```

**Comment by mistydemeo** [comment]
Thanks very much for this! I'd love to see this shipped sometime soon. I work on a package manager which is blocked on shipping 0.9.9.6 or 0.9.9.7 because there isn't a `make install` target and there isn't a documented way to install `glm`.

**Comment by Groovounet** [comment]
This PR was replaced by https://github.com/g-truc/glm/pull/966

**Comment by onodera-punpun** [comment]
No it wasn't, because #966 doesn't provide any way to install the package.

**Pull request closed without merging**